### ISSUE TITLE
Issue 06: `proc/train_mae.py --resume` と MAE checkpoint resume を実装する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#475]
# Issue 06: `proc/train_mae.py --resume` と MAE checkpoint resume を実装する

## 背景

Phase 7 の MAE pretraining は checkpoint 保存まではあるが、長時間実行では `--resume checkpoint.pt` が必要である。Issue 05 で checkpoint payload に `global_step`, `amp_enabled`, `scaler_state_dict` が入る前提で、ここでは training runtime に復元処理を追加する。

## 目的

`proc/train_mae.py --resume <checkpoint.pt>` を追加し、model / optimizer / scaler / global_step / epoch を復元して MAE pretraining を継続できるようにする。

## 依存 issue

- Issue 05: checkpoint state schema hardening

## 変更対象

- 更新: `proc/train_mae.py`
- 更新: `src/seis_attr_ssl/training/mae.py`
- 更新: `src/seis_attr_ssl/training/checkpoint.py` は helper が必要な場合のみ
- 更新: `tests/test_mae_training_engine.py`
- 更新: `tests/test_proc_train_mae_integration.py`
- 更新: `tests/test_proc_dry_run.py`

## CLI 仕様

`proc/train_mae.py` に引数を追加する。

```bash
python proc/train_mae.py \
  --config proc/configs/mvp_mae.yaml \
  --resume runs/mae_epoch_0003.pt
```

仕様:

- `--resume` は checkpoint path。
- path が存在しなければ non-zero exit。
- `--dry-run --resume path` では config validation と resume path existence check を行い、実際の checkpoint load / training は行わない。
- stdout の dry-run summary に `resume: <path>` 相当を出す。

## Runtime 仕様

`run_mae_pretraining()` の signature を拡張する。

```python
def run_mae_pretraining(
    config: Mapping[str, object],
    *,
    resume: str | Path | None = None,
) -> Path: ...
```

復元手順:

1. config validation 後、通常通り dataset / dataloader / model / optimizer / scaler を構築する。
2. `resume` が指定されたら `load_checkpoint(resume, map_location=device)` で payload を読む。
3. `model.load_state_dict(payload['model_state_dict'])` を strict に実行する。
4. `optimizer.load_state_dict(payload['optimizer_state_dict'])` を実行する。
5. AMP enabled かつ payload に `scaler_state_dict` があれば `scaler.load_state_dict(...)` を実行する。
6. `start_epoch = int(payload['epoch']) + 1` とする。
7. `global_step = int(payload.get('global_step', 0))` とする。
8. 以後の epoch loop は `range(start_epoch, epochs + 1)` で回す。

`train.max_steps` の意味は「resume 前後を含む total optimizer steps」とする。例えば checkpoint の `global_step == 1000` で `--max-steps 1200` の場合、追加で 200 steps 実行して止まる。

## Config mismatch の扱い

最初の実装では、checkpoint 内 config と CLI config を自動 merge しない。CLI config を正とする。ただし最低限、以下の mismatch は明示的にエラーにする。

- checkpoint に `model_state_dict` がない。
- checkpoint に `optimizer_state_dict` がない。
- checkpoint の `stage` が存在し、`pretrain_mae` ではない。

model architecture mismatch は `model.load_state_dict()` の error をそのまま出してよい。

## epoch 内 resume について

この issue の MVP は epoch-boundary resume とする。checkpoint の `epoch` の次 epoch から再開する。DataLoader sampler の epoch 内位置復元はスコープ外。Issue 07 で step checkpoint を追加する場合も、まず global_step 継続を優先し、完全な sample replay は要求しない。

### unit / integration

1. tiny config で 1 epoch 実行し、checkpoint epoch 1 / global_step 1 を作る。
2. 同じ config の `train.epochs = 2` で `run_mae_pretraining(config, resume=ckpt)` を呼ぶ。
3. 出力 checkpoint が epoch 2 になり、`global_step == 2` になる。
4. optimizer state が復元されることを確認する。厳密な数値一致までは不要だが、`optimizer_state_dict` を load しても error が出ないこと。
5. CPU / AMP disabled で scaler が None でも resume できる。
6. missing resume path は CLI で non-zero exit。
7. `proc/train_mae.py --help` に `--resume` が表示される。

### CLI

heavy test で既存の `test_train_mae_proc_one_step_cpu_run_writes_checkpoint` と同等の synthetic config を使い、可能なら以下を追加する。

```bash
python proc/train_mae.py --config tiny.yaml --device cpu --max-steps 1 --output-root runs
python proc/train_mae.py --config tiny.yaml --device cpu --max-steps 2 --output-root runs --resume runs/mae_epoch_0001.pt
```

2 回目で `mae_epoch_0002.pt` が作られ、`global_step == 2` になること。

## 完了条件

- `--resume checkpoint.pt` で model / optimizer / scaler / epoch / global_step が復元される。
- resume 後の `max_steps` が total step として機能する。
- 既存の non-resume training path が壊れない。

## 実行コマンド

```bash
python -m pytest tests/test_mae_training_engine.py
python -m pytest tests/test_proc_dry_run.py
SEIS_ATTR_SSL_RUN_HEAVY_TESTS=1 python -m pytest tests/test_proc_train_mae_integration.py -m heavy
```

## スコープ外

- epoch 内 batch index の完全復元。
- distributed training resume。
- scheduler state。現状 scheduler がないため不要。
